test/mockupdb/test_cluster_time.py

```diff
@@ -166,6 +167,17 @@ def test_monitor(self):
         request.reply(reply)
         client.close()
 
+    def test_bulk_error(self):
+        def callback(client: MongoClient[dict]) -> None:
+            client.db.collection.bulk_write([InsertOne({}), InsertOne({})])
+
+        with self.assertRaises(OperationFailure) as context:
+            self.cluster_time_conversation(
+                callback,
+                [{"ok": 0, "errmsg": "mock error"}],
+            )
+        self.assertIn("$clusterTime", str(context.exception))
```

Comment:
This test is skipping the clusterTime asserts in cluster_time_conversation. We need to do:
```
        def callback(client: MongoClient[dict]) -> None:
            with self.assertRaises(OperationFailure):
                client.db.collection.bulk_write([InsertOne({}), InsertOne({})])

        self.cluster_time_conversation(
            callback,
            [{"ok": 0, "errmsg": "mock error"}],
        )
```

Reply:
Could you also add a similar test for client.bulk_write?

Reply:
Done and done